From 7b51b6d06842de4851b6c20b61dda2ead78e68ff Mon Sep 17 00:00:00 2001
From: Mika Laitio <lamikr@pilppa.org>
Date: Wed, 8 May 2024 13:48:21 -0700
Subject: [PATCH 1/5] Do not fail on install for missing kdb.bz2 file

Do not fail if kdb.bz2 file for some GPU does not exist because
these kdb files are not mandatory. Their function is to speed up the
first calculation and they can also be generated afterwards.

Signed-off-by: Mika Laitio <lamikr@pilppa.org>
---
 CMakeLists.txt | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 32d9a2e5b..d6c2db704 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -538,7 +538,11 @@ set(COMPONENT_LST gfx942kdb gfx90akdb gfx1030kdb gfx908kdb gfx906kdb gfx900kdb)
 
 if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.17)
     foreach(__file __component IN ZIP_LISTS KDB_BZ2_FILES COMPONENT_LST)
-        install_kdb(${KERNELS_SOURCE_DIR}/${__file} ${__component})
+        # Do not fail on error if the kdb.bz2 file does not exist.
+        # Those are not available on some consumer level gpu's but values can be calculated and stored afterwards.
+        if(EXISTS "${KERNELS_SOURCE_DIR}/${__file}")
+            install_kdb(${KERNELS_SOURCE_DIR}/${__file} ${__component})
+        endif()
     endforeach()
 else()
     # TODO: Upgrade minimum CMake to version 3.17+ and use IN ZIP_LISTS instead
@@ -547,7 +551,11 @@ else()
     foreach(__index RANGE ${__high})
         list(GET COMPONENT_LST ${__index} __component)
         list(GET KDB_BZ2_FILES ${__index} __file)
-        install_kdb(${KERNELS_SOURCE_DIR}/${__file} ${__component})
+        # Do not fail on error if the kdb.bz2 file does not exist.
+        # Those are not available on some consumer level gpu's but values can be calculated and stored afterwards.
+        if(EXISTS "${KERNELS_SOURCE_DIR}/${__file}")
+            install_kdb(${KERNELS_SOURCE_DIR}/${__file} ${__component})
+        endif()
     endforeach()
 endif()
 
-- 
2.45.2

